src/VisualStudio/Core/Def/Implementation/ProjectSystem/AnalyzerConfigDocumentAsSolutionItemHandler.cs

```diff
@@ -57,9 +58,23 @@ void IDisposable.Dispose()
 
         private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
         {
-            // Check if a new analyzer config document was added and we have a non-null DTE instance.
-            if (e.Kind != WorkspaceChangeKind.AnalyzerConfigDocumentAdded ||
-                _dte == null)
+            switch (e.Kind)
+            {
+                case WorkspaceChangeKind.SolutionAdded:
```

Comment:
Can you file a bug to have us add a test against the VisualStudioWorkspace in the open/close/open again we raise SolutionAdded the second time? Looking at the code this is what we do, but I'm not sure if we have a test that would prevent us from breaking this.

Reply:
Filed https://github.com/dotnet/roslyn/issues/39476